filebeat/input/filestream/internal/input-logfile/harvester.go

```diff
@@ -236,6 +260,20 @@ func startHarvester(
 		cursor := makeCursor(resource)
 		publisher := &cursorPublisher{canceler: ctx.Cancelation, client: client, cursor: &cursor}
 
+		defer func() {
+			// The cursor struct used by Filestream, it is defined on:
+			// filebeat/input/filestream/input.go
+			st := struct {
+				Offset int64 `json:"offset" struct:"offset"`
+			}{}
+			if err := cursor.Unpack(&st); err != nil {
+				ctx.Logger.Errorf("cannot unpack cursor at the end of the harvester: %s", err)
+			}
+
+			hg.notifyObserver(srcID, st.Offset)
```

Comment:
If for some reason the Unpack didn't work, is it still safe to notify the observer with an offset of 0?  I'm guessing if the Unpack didn't work something went horribly wrong and we want to start from the beginning but just want to double check.

Reply:
Done 08f07f1bac.